.githooks/pre-push

```diff
@@ -1,31 +1,28 @@
 #!/bin/sh
 
-# An example hook script to verify what is about to be pushed.  Called by "git
-# push" after it has checked the remote status, but before anything has been
-# pushed.  If this script exits with a non-zero status nothing will be pushed.
-#
-# This hook is called with the following parameters:
-#
-# $1 -- Name of the remote to which the push is being done
-# $2 -- URL to which the push is being done
-#
-# If pushing without using a named remote those arguments will be equal.
-#
-# Information about the commits which are being pushed is supplied as lines to
-# the standard input in the form:
-#
-#   <local ref> <local oid> <remote ref> <remote oid>
+# Pre-push hook script to run code quality checks and ensure consistency.
 #
+# This script will execute two custom scripts located in the `scripts/` directory:
+# 1. clippy-on-all-workspaces.sh: Runs Clippy, tests, and formatting on all specified workspaces.
+# 2. sv2-header-check.sh: Ensures the `sv2.h` file generated by `build_header.sh` matches the
+#    committed version.
 
+# Exit immediately if any command exits with a non-zero status and print each command before
+# executing it.
 set -xe
 
-remote="$1"
-url="$2"
-
-act --job message_generator_check --reuse -P ubuntu-latest=ghcr.io/catthehacker/ubuntu:rust-latest
-act --job sv2_header_check --reuse -P ubuntu-latest=ghcr.io/catthehacker/ubuntu:rust-latest
-act --job fmt --reuse -P ubuntu-latest=ghcr.io/catthehacker/ubuntu:rust-latest
-act --job clippy-check --reuse -P ubuntu-latest=ghcr.io/catthehacker/ubuntu:rust-latest
-act --job ci --reuse -P ubuntu-latest=ghcr.io/catthehacker/ubuntu:rust-latest
+# Run clippy, test, and rustfmt on all workspaces
+sh ./scripts/clippy-on-all-workspaces.sh
```

Comment:
As suggested [here](https://github.com/stratum-mining/stratum/issues/1038#issuecomment-2211651871), I would rename the script into `clippy-fmt-and-test.sh`

Reply:
now that we removed cargo version from the toml files we should enforce cargo 1.75 in the scripts